cpp/json_ffi/config.cc

```diff
@@ -142,6 +144,65 @@ std::optional<std::vector<Data>> Conversation::AsPrompt(std::string* err) {
           }
         }
         message += role_text;
+      } else if (item["type"] == "image_url") {
+        if (item.find("image_url") == item.end()) {
+          *err += "Content should have an image_url field";
+          return std::nullopt;
+        }
+        std::string image_url =
+            item.at("image_url");  // TODO(mlc-team): According to OpenAI API reference this
+                                   // should be a map, with a "url" key containing the URL, but
+                                   // we are just assuming this as the URL for now
+        std::string base64_image = image_url.substr(image_url.find(",") + 1);
+        int width, height, num_channels;
+        auto image_data = LoadImageFromBase64(base64_image, &width, &height, &num_channels, err);
+        if (!image_data) {
+          return std::nullopt;
+        }
+
+        picojson::object config = json::LoadJSONFromString(model_config_str, err).value();
+        if (config.find("model_config") == config.end()) {
```

Comment:
Seems we are lazily loading these config per request. I think it is better to populate once during reload